pkg/reconciler/v1alpha1/revision/resolve.go

```diff
@@ -32,6 +34,35 @@ type digestResolver struct {
 	transport http.RoundTripper
 }
 
+const (
+	// Kubernetes CA certificate bundle is mounted into the pod here, see:
+	// https://kubernetes.io/docs/tasks/tls/managing-tls-in-a-cluster/#trusting-tls-in-a-cluster
+	k8sCertPath = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+)
+
+// newResolverTransport returns an http.Transport that appends the certs bundle
+// at path to the system cert pool.
+//
+// Use this with k8sCertPath to trust the same certs as the cluster.
+func newResolverTransport(path string) (*http.Transport, error) {
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		pool = x509.NewCertPool()
+	}
+
+	if crt, err := ioutil.ReadFile(path); err != nil {
+		return nil, err
+	} else if ok := pool.AppendCertsFromPEM(crt); !ok {
+		return nil, fmt.Errorf("Failed to append k8s cert bundle to cert pool.")
```

Comment:
`errors.New()` (you don't have a format string)

Reply:
Nice catch, thanks.